File: app/src/main/java/com/gigforce/app/utils/ThumbTextView.java
package com.gigforce.app.utils;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;
import android.widget.SeekBar;

import com.gigforce.app.R;

public class ThumbTextView extends androidx.appcompat.widget.AppCompatTextView {

    private LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT);
    private int width = 0;

    public ThumbTextView(Context context) {
        super(context);
    }

    public ThumbTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public void attachToSeekBar(SeekBar seekBar) {
        float contentWidth = this.getPaint().measureText(getText().toString());

        int width = seekBar.getWidth()
                - seekBar.getPaddingLeft()
                - seekBar.getPaddingRight();
        int thumbPos = seekBar.getPaddingLeft()
                + width
                * seekBar.getProgress()
                / seekBar.getMax();

        LinearLayout.LayoutParams layoutParams = (LinearLayout.LayoutParams) getLayoutParams();
        int left = getResources().getDimensionPixelSize(R.dimen.size_4);

        layoutParams.setMargins(thumbPos -left, 0, 0, 0);
        setLayoutParams(layoutParams);

    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        if (width == 0)
            width = MeasureSpec.getSize(widthMeasureSpec);
    }
}

File: app/src/main/java/com/gigforce/app/utils/ThumbTextView.kt
package com.gigforce.app.utils

import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout
import android.widget.SeekBar
import androidx.appcompat.widget.AppCompatTextView
import com.gigforce.app.R


class ThumbTextView : AppCompatTextView {
    private val lp = LinearLayout.LayoutParams(
        LinearLayout.LayoutParams.WRAP_CONTENT,
        LinearLayout.LayoutParams.WRAP_CONTENT
    )


    constructor(context: Context?) : super(context!!) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(
        context!!, attrs
    ) {
    }

    fun attachToSeekBar(seekBar: SeekBar) {
        val contentWidth = this.paint.measureText(text.toString())

        val width = (seekBar.width
                - seekBar.paddingLeft
                - seekBar.paddingRight)
        val thumbPos = (seekBar.paddingLeft
                + width
                * seekBar.progress
                / seekBar.max)
        val layoutParams = layoutParams as LinearLayout.LayoutParams
        val left = resources.getDimensionPixelSize(R.dimen.size_4)
        val top = resources.getDimensionPixelSize(R.dimen.size_9)
        layoutParams.setMargins(
            (thumbPos - (left + ((contentWidth - left) / 2))).toInt(),
            top,
            0,
            0
        )
        setLayoutParams(layoutParams)


    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        if (width == 0)
            width = MeasureSpec.getSize(widthMeasureSpec)
    }
}